python/cudf/cudf/core/df_protocol.py

```diff
@@ -0,0 +1,825 @@
+import collections
+import enum
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    cast,
+)
+
+import cupy as cp
+import numpy as np
+from numba.cuda import as_cuda_array
+
+import cudf
+from cudf.core.buffer import Buffer
+from cudf.core.column import as_column, build_categorical_column, build_column
+
+# Implementation of interchange protocol classes
+# ----------------------------------------------
+
+
+class _DtypeKind(enum.IntEnum):
+    INT = 0
+    UINT = 1
+    FLOAT = 2
+    BOOL = 20
+    STRING = 21  # UTF-8
+    DATETIME = 22
+    CATEGORICAL = 23
+
+
+class _Device(enum.IntEnum):
+    CPU = 1
+    CUDA = 2
+    CPU_PINNED = 3
+    OPENCL = 4
+    VULKAN = 7
+    METAL = 8
+    VPI = 9
+    ROCM = 10
+
+
+_k = _DtypeKind
+_SUPPORTED_KINDS = (
+    _k.INT,
+    _k.UINT,
+    _k.FLOAT,
+    _k.CATEGORICAL,
+    _k.BOOL,
+    _k.STRING,
+)
```

Comment:
Use a set, and eliminate the unnecessary `_k` variable here. The names are concise enough without the alias.
```suggestion
_SUPPORTED_KINDS = {
    _DtypeKind.INT,
    _DtypeKind.UINT,
    _DtypeKind.FLOAT,
    _DtypeKind.CATEGORICAL,
    _DtypeKind.BOOL,
    _DtypeKind.STRING,
}
```